src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/TextBlockGoogleStyleFormattingCheck.java

```diff
@@ -0,0 +1,203 @@
+ * </p>
+ *
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ *
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code textblock.format.close}
+ * </li>
+ *
+ * <li>
+ * {@code textblock.format.open}
+ * </li>
+ *
+ * <li>
+ * {@code textblock.indentation}
+ * </li>
+ * </ul>
+ *
+ * @since 11.1.0
+ */
+@StatelessCheck
+public class TextBlockGoogleStyleFormattingCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_OPEN_QUOTES_ERROR = "textblock.format.open";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_CLOSE_QUOTES_ERROR = "textblock.format.close";
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_INDENTATION_ERROR = "textblock.indentation";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.TEXT_BLOCK_LITERAL_BEGIN,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (!openingQuotesAreOnCorrectPosition(ast)) {
+            log(ast, MSG_OPEN_QUOTES_ERROR);
+        }
+
+        final DetailAST closingQuotes = getClosingQuotes(ast);
+        if (!closingQuotesAreOnCorrectPosition(closingQuotes)) {
+            log(closingQuotes, MSG_CLOSE_QUOTES_ERROR);
+        }
+
+        if (!isVerticallyAligned(ast, closingQuotes)) {
+            log(closingQuotes, MSG_INDENTATION_ERROR);
+        }
+    }
+
+    /**
+     * Checks if opening and closing quotes are vertically aligned.
+     *
+     * @param openQuotes the ast to check.
+     * @param closeQuotes the ast to check.
+     * @return true if both quotes have same indentation else false.
+     */
+    private static boolean isVerticallyAligned(DetailAST openQuotes, DetailAST closeQuotes) {
+        return openQuotes.getColumnNo() == closeQuotes.getColumnNo();
+    }
+
+    /**
+     * Gets the {@code TEXT_BLOCK_LITERAL_END} of a {@code TEXT_BLOCK_LITERAL_BEGIN}.
+     *
+     * @param ast the ast to check
+     * @return DetailAST {@code TEXT_BLOCK_LITERAL_END}
+     */
+    private static DetailAST getClosingQuotes(DetailAST ast) {
+        return ast.getFirstChild().getNextSibling();
+    }
+
+    /**
+     * Determines if the Opening quotes of text block are not preceded by any code.
+     *
+     * @param openingQuotes opening quotes
+     * @return true if the opening quotes are on the new line.
+     */
+    private static boolean openingQuotesAreOnCorrectPosition(DetailAST openingQuotes) {
```

Comment:
```suggestion
    private static boolean openingQuotesAreAloneOnTheLine(DetailAST openingQuotes) {
```

This name is bit misleading. We're checking that opening quotes are not preceded by any other code. So the method should be renamed.